server/service/src/stocktake_line/insert.rs

```diff
@@ -164,6 +183,23 @@ fn validate(
         }
     }
 
+    let stocktake_difference = stocktake_difference(&input.counted_number_of_packs, &stock_line);
+    if check_active_adjustment_reasons(connection, stocktake_difference)?.is_some()
+        && input.inventory_adjustment_reason_id.is_none()
+        && stocktake_difference != 0.0
```

Comment:
Or 
```suggestion
        && input.counted_number_of_packs.is_none()
```
🤷🏻

Reply:
the two lines aren't equivalent 😕 - I would run with `stocktake_difference != 0.0` that makes sense to me